Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm

```diff
@@ -2149,7 +2150,7 @@ static void fulfillRequestWithKeyData(AVAssetResourceLoadingRequest *request, Ar
         auto initData = SharedBuffer::create(Vector<uint8_t> { static_cast<uint8_t*>(initDataBuffer->data()), byteLength });
         player->keyNeeded(initData);
 #if ENABLE(ENCRYPTED_MEDIA)
-        if (!player->shouldContinueAfterKeyNeeded())
+        if (!m_shouldContinueAfterKeyNeeded)
```

Comment:
querying the player is better I believe, it's also simpler. and would remove all the changes to the MediaPlayerPrivate, so long as the respective MediaPlayer is up to do, it's sufficient.

Reply:
Different sure, but why "better"? `setShouldContinueAfterKeyNeeded` is virtual presumably so it can be overridden. See my comment below about why I added the override.